src/db/db.py

```diff
@@ -22,6 +22,21 @@ def _api_key_valid(session: Session, api_key: str):
     """ Check the API key exists and is enabled """
     return session.scalars(select(DbApiKey).filter_by(api_key=api_key, enabled=True)).first() is not None
 
+def _api_key_write_enabled(session: Session, api_key: str):
+    """ Check the API key exists and is enabled """
+    return session.scalars(select(DbApiKey).filter_by(api_key=api_key, enabled=True, write_enabled=True)).first() is not None
+
+
+def add_article(article: DbArticle, api_key: str) -> DbArticle:
+    """ Add an article to the database if given a valid API key """
+    with DbSession() as session:
+        if not _api_key_write_enabled(session, api_key):
+            raise HTTPException(status_code=403, detail="Invalid API key")
+        article.author = api_key
```

Comment:
Can we call this something else? "author" feels to me like it should refer to the _content_ of the document -- in this case, `registrant` feels more appropriate

Reply:
That makes sense, I'll update the form name